### ISSUE TITLE
[Enhancement]: Ignore Synology @eaDir folder and @SynoEAStream files in Library Files

### ISSUE THREAD

**Issue by hypnotoad08** [advplyr/audiobookshelf#1641]
### Describe the feature/enhancement

This is a minor feature request as it doesn't affect functionality. I noticed that the Library files section for some of the audiobook files I have moved to my shared Audiobook folder on my Synology NAS are showing some hidden files/folders that are created by the NAS. Can the Library files section filter out these hidden files/folders?
![@eaDir](https://user-images.githubusercontent.com/842086/229292817-d4e041a1-f7dd-4419-9813-f40e12d96c2a.jpg)

**Comment by CLHatch** [comment]
This is handled by Plex with `.plexignore` files that can be created to specify patterns of files/folders to be ignored.  Something similar could be created for ABS. https://support.plex.tv/articles/201381883-special-keyword-file-folder-exclusion/

**Comment by CLHatch** [comment]
SyncThing has a similar file you can use to set files/folders to ignore.

**Comment by advplyr** [comment]
You can use a `.ignore` file

**Comment by raphpa** [comment]
It is a little bit more complicated with the `@eaDir` folder. The NAS creates it automatically in every folder with media in it. So if you have your library sorted in subfolders you would have to manually put the `.ignore` file manually into every single one of them.
A global ignore option would be great for this case.

**Comment by hypnotoad08** [comment]
Yes I was hoping for a global fix

**Comment by theolike** [comment]
Agreed.  It has to be a configurable global filter.  While this one is specifically a Synology quirk, there are many other systems that use a similar process for indexing, thumbnails, etc.  For example, Windows and it's thumbs.db files.  I haven't been through the indexer code yet (and I guess I should do that), but is there already filter logic in it, even if for only a hard coded list?

(Also, no matter what kind of spoon you are, "Journey before destination", Radiant.)

**Comment by CLHatch** [comment]
> Agreed. It has to be a configurable global filter. While this one is specifically a Synology quirk, there are many other systems that use a similar process for indexing, thumbnails, etc. For example, Windows and it's thumbs.db files. I haven't been through the indexer code yet (and I guess I should do that), but is there already filter logic in it, even if for only a hard coded list?
> 
> (Also, no matter what kind of spoon you are, "Journey before destination", Radiant.)

Yeah, I actually just had to add an "ignore" option for one other folder in my SyncThing shares.  I noticed that `.@__thumb` folders kept appearing in folders.  It contained thumbnails of various sizes.  Ends up, someone else in the share has a QNAP, it creates them. Added `(?d).@__thumb`, and all good (the `(?d)` tells SyncThing it's ok to delete the files if it's needed to delete the parent folder.  Obviously flags like that aren't needed for ABS.)

**Comment by DDriggs00** [comment]
Additionally, when deleting an item from the library only (not filesystem), it would be useful to add an option to create an ignore file on the offending directory.

**Comment by iconoclasthero** [comment]
mpd also has this and it would be useful to have .absignore similar to .mpdignore

**Comment by nichwall** [comment]
Doing a bit of research tonight looks like `node-ignore` may be a useful package for adding an ignore list. Just putting this here for reference. I couldn't find many resources for it (just kept getting references of how to use gitignore, plexignore, etc, not actual implementations)

https://github.com/kaelzhang/node-ignore

**Comment by rage1337** [comment]
Found a workaround for my use case (described [here):](https://github.com/advplyr/audiobookshelf/issues/2460)
This approach helps me to exclude my workfolder: [Stackoverflow](https://stackoverflow.com/a/37898591/5097619)

Changed 
`volumes:
      - /vault/media/audiobooks:/audiobooks
      - /vault/media/podcasts:/podcasts
      - ./config:/config
      - ./data:/metadata
`
to
`volumes:
      - /vault/media/audiobooks:/audiobooks
      - /vault/media/podcasts:/podcasts
      - /audiobooks/_
      - ./config:/config
      - ./data:/metadata
`
The folder is there, but empty inside the container.

**Comment by iconoclasthero** [comment]
I think i put a FR in for this.  I don't know what's going on above, but I could use this right now.  Also, I'd want to behave the way that .mpdignore behaves so I could put `tmp` in .absignore in the root of the library and be done with it.

**Comment by chinajuanbob** [comment]
Any updates or solution for this? Thx!

**Comment by nichwall** [comment]
2.19.5 updated the ignore list. There is still not a user configurable option, but the framework is there to more easily add vendor specific folders and extensions.

https://github.com/advplyr/audiobookshelf/pull/4031

**Comment by SAS-1** [comment]
Nice one kicked off a scan and all the eadir is gone.

**Comment by thrandale** [comment]
It's better now, but I'd still love to see the ability to specify a `.ignore` file in the base library directory, with the contents being what I'd like to ignore. Following the format of something like a `.gitignore`. Some example usecases:
1. On a Ugreen nas, the recycle folder is `#recycle` not `.recycle`, meaning it isn't ignored.
2. When riping audiobooks, the program I am using creates `temp....wav` files that I would like to be able to ignore.
